### PR TITLE
?snipe cmd

### PR DESCRIPTION
**Pull request by aarctan**
**Need audit log perms to work**
Asked jack ma for audit log perm on %

### CODE REVIEW

File: bot.py [hunk 1]
```diff
@@ -40,6 +40,20 @@
...
 
+deleted_messages = {} # {channel_id : message}
+
+@client.event
+async def on_message_delete(message):
+    deleted_messages[message.channel.id] = message
+
+@client.command(name='snipe')
+async def snipe(ctx):
+    message = deleted_messages.pop(ctx.channel.id, None)
```
Comment (Mattwmaster58) [inline bot.py]: do we only want snipe to work on first invocation?

Comment (aarctan) [reply bot.py]: we shouldn't maintain a list of deleted messages. some people may want to erase their old messages. the purpose for only sniping the last deleted message is for seeing what someone deleted right as it was deleted. it'd be even better to only maintain that message in this dict for just a few seconds.

File: bot.py [hunk 2]
```diff
@@ -40,6 +41,28 @@
 with open("knowledge/specific.json", "r", encoding='utf-8') as f:
     kattis_specific = json.load(f)
 
+##### ?snipe
+class DeletedMsg:
```
Comment (Mattwmaster58) [inline bot.py:45]: use data class to be more pythonic

File: bot.py [hunk 3]
```diff
@@ -40,6 +41,28 @@
...
+        self.msg = msg
+        self.time = datetime.utcnow()
+
+deleted_messages = {} # {channel id : DeletedMsg}
+
+@client.event
+async def on_message_delete(message):
+    deleted_messages[message.channel.id] = DeletedMsg(message)
+
+@client.command(name='snipe')
```
Comment (Mattwmaster58) [inline bot.py:56]: Slash commands are slow to invoke (that's what this is right?) Can we use the legacy thingy

**Review by Mattwmaster58 — APPROVED** [review]

**Pull request merged**